ctru-rs/src/services/ac.rs

```diff
@@ -0,0 +1,347 @@
+    ///
+    /// println!("Proxy port: {}", ac.get_proxy_port()?);
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetProxyPort")]
+    pub fn get_proxy_port(&self) -> crate::Result<u32> {
+        unsafe {
+            let mut ret = 0u32;
+            ResultCode(ctru_sys::ACU_GetProxyPort(&mut ret))?;
+
+            Ok(ret)
+        }
+    }
+
+    /// Returns the connected network's proxy username, if present.
+    ///
+    /// You can check if the console is using a proxy with [`Self::get_proxy_enabled()`]
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Proxy username: {}", ac.get_proxy_username()?);
+    ///
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetProxyUserName")]
+    pub fn get_proxy_username(&self) -> crate::Result<String> {
+        unsafe {
+            let mut vec = vec![0u8; 0x20];
+            ResultCode(ctru_sys::ACU_GetProxyUserName(vec.as_mut_ptr()))?;
+
+            // how do i handle this error?
+            Ok(String::from_utf8(vec)?)
+        }
+    }
+
+    /// Returns the connected network's proxy password, if present.
+    ///
+    /// You can check if the console is using a proxy with [`Self::get_proxy_enabled()`]
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Proxy password: {}", ac.get_proxy_password()?);
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetProxyPassword")]
+    pub fn get_proxy_password(&self) -> crate::Result<String> {
+        unsafe {
+            let mut vec = vec![0u8; 0x20];
+            ResultCode(ctru_sys::ACU_GetProxyPassword(vec.as_mut_ptr()))?;
+
+            // how do i handle this error?
+            Ok(String::from_utf8(vec)?)
+        }
+    }
+
+    /// Load the selected network slot, if present.
+    ///
+    /// Note: this method requires `ac:i` access
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// ac.load_network_slot(NetworkSlot::Second)?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACI_LoadNetworkSetting")]
+    pub fn load_network_slot(&self, slot: NetworkSlot) -> crate::Result<()> {
+        unsafe {
+            ResultCode(ctru_sys::ACI_LoadNetworkSetting(slot as u32))?;
```

Comment:
Hmm, I guess this API is not available yet in the base `devkitpro/devkitarm` Docker image we use for CI, which apparently only has `libctru 2.2.2-2` installed.

I think this could be resolved a couple different ways, offhand
- We could always update to latest devkitPro in CI. Might not work since IIRC the devkitPro maintainers block downloads from Github Actions runner IPs...
- We could do a little work to add a `#[cfg]` for this based on the `libctru` 
version (easier once #141 is merged)
- Wait for a new docker image to be published... These come every couple months normally but not sure how long it would have to wait. It could be commented out or `#[cfg(ignore)]` or something in the meantime just to get this PR merged
- Fork and maintain our own docker image 🤮 not my preference this one

@rust3ds/active any thoughts?

Reply:
the `#[cfg]` based on version is the cleanest way, yes, but I don't see any meaningful reason to support multiple `libctru` versions at the same time (especially on something like this, if anything)

so yeah

Reply:
There is a new devkitARM docker image as of `20240202` so I think this is good (and linter passed)